tensorflow/python/eager/function_test.py

```diff
@@ -3621,8 +3621,15 @@ def func(x, kangaroo=None, octopus=7):
                   r'      <1>: int32 Tensor, shape=\(\)\n'
                   r'      <2>: RaggedTensorSpec\(.*\)\n'
                   r'      <3>: RaggedTensorSpec\(.*\)')
-    self.assertRegexpMatches(c3.pretty_printed_signature(),
-                             c3_summary + '\n' + c3_details)
+
+    # The output of pretty_printed_signature is not deterministic for dict
```

Comment:
I don't follow shouldn't this follow insertion order? At least on python3? What is the ROCm-specific issue here?

Reply:
This is not a ROCM specific issue per-se.  We just saw this subtest fail a couple of times on the ROCm CI, and hence need to disable it atleast for ROCm.

Here is the error we get when this test fails

```
======================================================================                                                                                          
FAIL: testPrettyPrintedSignature (__main__.FunctionTest)                                                                                                        
testPrettyPrintedSignature (__main__.FunctionTest)                                                                                                              
----------------------------------------------------------------------                                                                                          
Traceback (most recent call last):                                                                                                                              
  File "/home/jenkins/workspace/tensorflow-rocm-nightly/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorfl\
ow/bazel-out/k8-opt/bin/tensorflow/python/eager/function_test_gpu.runfiles/org_tensorflow/tensorflow/python/eager/function_test.py", line 3625, in testPrettyPr\
intedSignature                                                                                                                                                  
    c3_summary + '\n' + c3_details)                                                                                                                             
AssertionError: Regex didn't match: "func\\(x, kangaroo=None, octopus=7\\)\n  Args:\\n    x: {'a': <1>, 'b': \\[<2>, <3>\\]}\\n      <1>: int32 Tensor, shape=\\
\(\\)\\n      <2>: RaggedTensorSpec\\(.*\\)\\n      <3>: RaggedTensorSpec\\(.*\\)\\n  Returns:\\n    {'a': <1>, 'b': \\[<2>, <3>\\]}\\n      <1>: int32 Tensor,\
 shape=\\(\\)\\n      <2>: RaggedTensorSpec\\(.*\\)\\n      <3>: RaggedTensorSpec\\(.*\\)" not found in "func(x, kangaroo=None, octopus=7)\n  Args:\n    x: {'b\
': [<2>, <3>], 'a': <1>}\n      <1>: int32 Tensor, shape=()\n      <2>: RaggedTensorSpec(TensorShape([2, None]), tf.int32, 1, tf.int64)\n      <3>: RaggedTenso\
rSpec(TensorShape([2, None]), tf.int32, 1, tf.int64)\n  Returns:\n    {'b': [<2>, <3>], 'a': <1>}\n      <1>: int32 Tensor, shape=()\n      <2>: RaggedTensorSp\
ec(TensorShape([2, None]), tf.int32, 1, tf.int64)\n      <3>: RaggedTensorSpec(TensorShape([2, None]), tf.int32, 1, tf.int64)"                                  
```

As you can see, the output of pretty_printed_signature can print `b` before `a`!  My understanding is that that routine simply iterates over the dictionary items and prints them. What is the gurantee that `a` will come become `b` in the iteration?

Reply:
think ROCm CI uses containers that are Python 3.5 based, and deterministic ordering for dictionary keys is in Ptyhont 3.6+ (thank you @ekuznetsov139  for researching this)

https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6

Reply:
@jaingaurav , 

updated the subtest skip to be Python 3.5 specific (instead of being ROCm specific).  Thanks to @ekuznetsov139 for the change.

please re-review. thanks

Reply:
@jaingaurav gentle ping

Reply:
Thanks. Could you please update the check to `sys.version_info >= (3, 5):` with attention to the whitespaces.

Reply:
@jaingaurav done